android/src/com/frostwire/android/offers/Offers.java

```diff
@@ -59,6 +60,10 @@ public static void initAdNetworks(Activity activity) {
         for (AdNetwork adNetwork : AD_NETWORKS) {
             adNetwork.initialize(activity);
         }
+
+        // I do this afterwards because the Products.listEnabled() method shouldn't be done on the UI thread.
+        // also the network.stop() method shouldn't have any thread-safety issues.
+        stopAdNetworksIfPurchasedRemoveAds(activity);
```

Comment:
The problematic method is `PlayStore#consumePurchasedProduct` and is used only if `INTERNAL_BUILD`, `Products.listEnabled` is safe in the UI, no need of a thread here